sqlx-core/src/mysql/types/json.rs

```diff
@@ -0,0 +1,58 @@
+use crate::decode::Decode;
+use crate::encode::Encode;
+use crate::mysql::database::MySql;
+use crate::mysql::protocol::TypeId;
+use crate::mysql::types::*;
+use crate::mysql::{MySqlTypeInfo, MySqlValue};
+use crate::types::{Json, Type};
+use serde::{Deserialize, Serialize};
+use serde_json::Value as JsonValue;
+
+impl Type<MySql> for JsonValue {
+    fn type_info() -> MySqlTypeInfo {
+        <Json<Self> as Type<MySql>>::type_info()
+    }
+}
+
+impl Encode<MySql> for JsonValue {
+    fn encode(&self, buf: &mut Vec<u8>) {
+        (Box::new(Json(self)) as Box<dyn Encode<MySql>>).encode(buf)
```

Comment:
I don't think I understand the rust type system well enough yet, is this the proper way to handle this?

Reply:
```rust
Json(self).encode(buf)
```

^ That should be enough, same as https://github.com/launchbadge/sqlx/blob/master/sqlx-core/src/postgres/types/json.rs#L26

In fact.. we probably should generalize the implementations of `JsonValue` and `JsonRawValue` and move them to a `src/types/json.rs`. They should be identical if the database driver supports `Json<_>`.